lib/Frontend/CompilerInvocation.cpp

```diff
@@ -928,12 +941,55 @@ static bool ParseDiagnosticArgs(DiagnosticOptions &Opts, ArgList &Args,
   if (Arg *A = Args.getLastArg(OPT_diagnostic_documentation_path)) {
     Opts.DiagnosticDocumentationPath = A->getValue();
   }
+  if (Arg *A = Args.getLastArg(OPT_localization_path)) {
+    if (!llvm::sys::fs::exists(A->getValue()))
+      Diags.diagnose(SourceLoc(), diag::locale_path_not_found, A->getValue());
+    else
+      Opts.LocalizationPath = A->getValue();
+  }
+  if (Arg *A = Args.getLastArg(OPT_locale)) {
+    Opts.LocalizationCode = A->getValue();
+  }
   assert(!(Opts.WarningsAsErrors && Opts.SuppressWarnings) &&
          "conflicting arguments; should have been caught by driver");
 
   return false;
 }
 
+static void ValidateLocalizationArgs(DiagnosticOptions &Opts, DiagnosticEngine &Diags) {
+
+  // Check if the locale code is available.
+  if (!Opts.LocalizationCode.empty()) {
+    std::string localeCode = Opts.LocalizationCode;
+
+    if (llvm::none_of(localeCodes, [&](const char *locale) {
+      return localeCode == locale;
+    })) {
+      std::string availableLocaleCodes = "";
+      llvm::interleave(
+                       std::begin(localeCodes), std::end(localeCodes),
+                       [&](std::string locale) { availableLocaleCodes += locale; },
+                       [&] { availableLocaleCodes += ", "; });
+      
+      // Incorrect locale code shouldn't crash the compiler,
+      // and the compiler will fallback to the default messages.
+      Diags.diagnose(SourceLoc(), diag::error_invalid_locale_code,
```

Comment:
One thing, the comment says that it should fall back to default messages, but as far as I can see this is an error, so this will just stop compilation here... Is this correct? @xedin

Reply:
Good catch! @HassanElDesouky we need this method to return `bool` so it's possible to `return true;` from `CompilerInvocation::parseArgs` if there is an error and localizations arguments are invalid.

Reply:
@xedin this `diag::error_invalid_locale_code` also should become a warning?

Reply:
No, I think that should stay an error.

Reply:
Looking at this once again I don't really see a reason for this validation function to exist, it's okay to get each diagnosed while argument is "parsed" just like the rest of `ParseDiagnosticArgs` has it.

Reply:
That would also make it clear what is a warning and what is an error.

Reply:
And also maintaining consistency with the other ones :)